presentation/src/main/java/me/androidbox/presentation/task/viewmodel/TaskDetailViewModel.kt

```diff
@@ -1,29 +1,97 @@
 package me.androidbox.presentation.task.viewmodel
 
+import android.util.Log
+import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.collectLatest
 import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import me.androidbox.data.local.agenda.TaskRepositoryImp
+import me.androidbox.domain.DateTimeFormatterProvider.toZoneDateTime
+import me.androidbox.domain.agenda.model.Task
 import me.androidbox.domain.alarm_manager.AlarmScheduler
+import me.androidbox.domain.alarm_manager.toAlarmItem
+import me.androidbox.domain.authentication.ResponseState
+import me.androidbox.presentation.agenda.constant.AgendaMenuActionType
+import me.androidbox.presentation.alarm_manager.AlarmReminderProvider
+import me.androidbox.presentation.navigation.Screen.Companion.MENU_ACTION_TYPE
+import me.androidbox.presentation.navigation.Screen.Companion.ID
 import me.androidbox.presentation.task.screen.TaskDetailScreenEvent
 import me.androidbox.presentation.task.screen.TaskDetailScreenState
+import java.util.UUID
 import javax.inject.Inject
 
 @HiltViewModel
 class TaskDetailViewModel @Inject constructor(
     private val alarmScheduler: AlarmScheduler,
+    private val taskRepositoryImp: TaskRepositoryImp,
```

Comment:
Never receive the Implementations
Only accept the abstractions

In this case
private val taskRepository: TaskRepository,

Reply:
@MKiperszmid Updated